### PR TITLE
Add support for MNK65_STM32

### PR DESCRIPTION
**Pull request by kopibeng**
## Description
Add support for MNK65_STM32

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kopibeng/mnk65_stm32/config.h [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+//#define VENDOR_ID       0x4B50 // 'KP' kopibeng  
+//#define PRODUCT_ID      0x0652
+//#define DEVICE_VER      0x0001
+//#define MANUFACTURER    kopibeng
+//#define PRODUCT         MNK65
```
Comment (drashna) [inline keyboards/kopibeng/mnk65_stm32/config.h:26]:
```suggestion
```

File: keyboards/kopibeng/mnk65_stm32/readme.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# MNK65
+
+![MNK65](https://i.imgur.com/YTrMDHj.jpg)
```
Comment (drashna) [inline keyboards/kopibeng/mnk65_stm32/readme.md:3]:
```suggestion
![MNK65](https://i.imgur.com/YTrMDHjh.jpg)
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/kopibeng/mnk65_stm32/mnk65.c [hunk 3]
```diff
@@ -0,0 +1,17 @@
+/* Copyright 2021 Samuel Lu
```
Comment (drashna) [inline keyboards/kopibeng/mnk65_stm32/mnk65.c:1]: File needs to match the folder name.  Eg, `mnk65_stm32.c`

File: keyboards/kopibeng/mnk65_stm32/mnk65.h [hunk 4]
```diff
@@ -0,0 +1,61 @@
+/* Copyright 2021 Samuel Lu
```
Comment (drashna) [inline keyboards/kopibeng/mnk65_stm32/mnk65.h:1]: File needs to match the folder name.  Eg, `mnk65_stm32.h`

File: keyboards/kopibeng/mnk65_stm32/mnk65.h [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+ *             │40  │41  │42  │44        │46  │48      │4A  │4B  │ 2.75u-1.25u-2.25u Split
+ *             └────┴────┴────┴──────────┴────┴────────┴────┴────┘
+ *             ┌─────┬───┬─────┬───────────────────────────┬─────┐
+ *             │40   │41 │42   │46                         │4B   │ 7u Tsangan
+ *             └─────┴───┴─────┴───────────────────────────┴─────┘
+ *
+ */
+
+// LAYOUT_all:
+#define LAYOUT_all( \
```
Comment (drashna) [inline keyboards/kopibeng/mnk65_stm32/mnk65.h:49]: If this is the only layout included here, then it should be just `LAYOUT`.
```suggestion
#define LAYOUT( \
```

However, you should probably break out the different layouts.

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/kopibeng/mnk65_stm32/readme.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+A QMK-powered, VIA-enabled 65% PCB with support for ANSI/ISO layouts, split Backspace and various bottom row configurations.
+
+* Keyboard Maintainer: kopibeng
+* Hardware Supported: MONOKEI
+* Hardware Availability: https://monokei.co/
+
+Make example for this keyboard (after setting up your build environment):
+
+    make kopibeng/mnk65_stm32:via
```
Comment (fauxpark) [inline keyboards/kopibeng/mnk65_stm32/readme.md:13]:
```suggestion
```

File: keyboards/kopibeng/mnk65_stm32/mnk65_stm32.h [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    K20, K21, K22, K23, K24, K25, K26, K27, K28, K29, K210, K211,       K213,       K214, \
+    K30, K31, K32, K33, K34, K35, K36, K37, K38, K39, K310, K311, K312,       K313, K314, \
+    K40, K41, K42,      K44,      K46,      K48,      K410, K411,       K412, K413, K414  \
+) { \
+    {K00, K01, K02, K03, K04, K05, K06, K07, K08, K09, K010, K011, K012, K013, K014}, \
+    {K10, K11, K12, K13, K14, K15, K16, K17, K18, K19, K110, K111, K112, K113, K114}, \
+    {K20, K21, K22, K23, K24, K25, K26, K27, K28, K29, K210, K211, K212, K213, K214}, \
+    {K30, K31, K32, K33, K34, K35, K36, K37, K38, K39, K310, K311, K312, K313, K314}, \
+    {K40, K41, K42, ___, K44, ___, K46, ___, K48, ___, K410, K411, K412, K413, K414}  \
+}
```
Comment (fauxpark) [inline keyboards/kopibeng/mnk65_stm32/mnk65_stm32.h:61]:
```suggestion
```
You have matrix positions in the info.json layout so this isn't needed.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/kopibeng/mnk65_stm32/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	// Default layer
+	[0] = LAYOUT(
+		KC_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_BSPC, KC_HOME,
```
Comment (fauxpark) [inline keyboards/kopibeng/mnk65_stm32/keymaps/via/keymap.c:23]:
```suggestion
		QK_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_BSPC, KC_HOME,
```

File: keyboards/kopibeng/mnk65_stm32/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	// Default layer
+	[0] = LAYOUT(
+		KC_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_BSPC, KC_HOME,
```
Comment (fauxpark) [inline keyboards/kopibeng/mnk65_stm32/keymaps/default/keymap.c:23]:
```suggestion
		QK_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_BSPC, KC_HOME,
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/kopibeng/mnk65_stm32/mnk65_stm32.h [hunk 10]
```diff
@@ -0,0 +1,47 @@
+/* Copyright 2021 Samuel Lu
```
Comment (fauxpark) [inline keyboards/kopibeng/mnk65_stm32/mnk65_stm32.h:1]: This file can be deleted.

File: keyboards/kopibeng/mnk65_stm32/mnk65_stm32.c [hunk 11]
```diff
@@ -0,0 +1,17 @@
+/* Copyright 2021 Samuel Lu
```
Comment (fauxpark) [inline keyboards/kopibeng/mnk65_stm32/mnk65_stm32.c:1]: This file can be deleted.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by tzarc — APPROVED** [review]

**Pull request merged**